dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityPreviewServlet.java

```diff
@@ -0,0 +1,359 @@
+                default:
+                    response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+                    return;
+            }
+
+
+
+        } catch (ResourceNotFoundException rnfe) {
+            Logger.error(this, "ResourceNotFoundException" + rnfe.toString(), rnfe);
+            response.sendError(404);
+            return;
+        } catch (ParseErrorException pee) {
+            Logger.error(this, "Template Parse Exception : " + pee.toString(), pee);
+            response.sendError(500, "Template Parse Exception");
+        } catch (MethodInvocationException mie) {
+            Logger.error(this, "MethodInvocationException" + mie.toString(), mie);
+            response.sendError(500, "MethodInvocationException Error on template");
+        } catch (Exception e) {
+            Logger.error(this, "Exception" + e.toString(), e);
+            response.sendError(500, "Exception Error on template");
+
+        } finally {
+            DbConnectionFactory.closeSilently();
+        }
+
+    }
+
+    public void init(ServletConfig config) throws ServletException {
+
+
+
+    }
+
+    public void doPreviewMode(HttpServletRequest request, HttpServletResponse response) throws Exception {
+
+        String uri = URLDecoder.decode(request.getRequestURI(), UtilMethods.getCharsetConfiguration());
+        PageMode mode = PageMode.PREVIEW_MODE;
+
+        Host host = hostWebAPI.getCurrentHost(request);
+
+
+        // Getting the user to check the permissions
+        User user = com.liferay.portal.util.PortalUtil.getUser(request);
+
+        // Getting the identifier from the uri
+        Identifier id = APILocator.getIdentifierAPI()
+            .find(host, uri);
+        request.setAttribute("idInode", id.getId());
+        Logger.debug(VelocityPreviewServlet.class, "VELOCITY HTML INODE=" + id.getId());
+
+        Template template = null;
+        Template hostVariablesTemplate = null;
+
+        // creates the context where to place the variables
+        response.setContentType(CHARSET);
+        Context context = VelocityUtil.getWebContext(request, response);
+
+        IHTMLPage htmlPage = VelocityUtil.getPage(id, request, false, context);
+        if ("contentlet".equals(htmlPage.getType())) {
+            context.put("dotPageContent", new ContentMap(((Contentlet) htmlPage), user, true, host, context));
+        }
+
+
+
+        com.dotmarketing.portlets.templates.model.Template cmsTemplate = APILocator.getHTMLPageAssetAPI()
+            .getTemplate(htmlPage, true);
+
+
+
+        VelocityUtil.makeBackendContext(context, htmlPage, cmsTemplate.getInode(), id.getURI(), request, mode, host);
+
+
+
+        PageContextBuilder builder = new PageContextBuilder(htmlPage, user, PageMode.PREVIEW_MODE);
+        builder.addAll(context);
+
+
+
+        hostVariablesTemplate = VelocityUtil.getEngine()
+            .getTemplate(PageMode.PREVIEW_MODE.name() + File.separator + host.getIdentifier() + "." + VelocityType.SITE.fileExtension);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add a "try/catch" block for "getTemplate". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1989)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add a "try/catch" block for "getTemplate". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1989)